docs/guide/obs-integrations.asciidoc

```diff
@@ -0,0 +1,179 @@
+In some supported frameworks, this is already done for you.
+In other scenarios, like for unstructured logs,
+you'll need to add APM identifiers to your logs in any easy to parse manner.
+
+The identifiers we're interested in are: {ecs-ref}/ecs-tracing.html[`trace.id`] and
+{ecs-ref}/ecs-tracing.html[`transaction.id`]. Certain Agents also support the `span.id` field.
+
+This process for adding these fields will differ based the Agent you're using, the logging framework, 
+and the type and structure of your logs.
+
+See the relevant Agent documentation to learn more:
+
+* *Go*: {apm-go-ref-v}/supported-tech.html#supported-tech-logging[Logging frameworks]
+* *Java*: {apm-java-ref-v}/config-logging.html#config-enable-log-correlation[`enable_log_correlation`]
+// * *.NET*: {apm-dotnet-ref-v}/[]
+// * *Node.js*: {apm-node-ref-v}/[]
+// * *Python*: {apm-py-ref-v}/[]
+// * *Ruby*: {apm-ruby-ref-v}/[]
+// * *Rum*: {apm-rum-ref-v}/[]
+
+[float]
+==== Ingest your logs into Elasticsearch
+
+Once your logs contain the appropriate identifiers (fields), you need to ingest them into Elasticsearch.
+Luckily, we've got a tool for that -- Filebeat is Elastic's log shipper.
+The {filebeat-ref}/filebeat-getting-started.html[Filebeat getting started]
+guide will walk you through the setup process.
+
+Because logging frameworks and formats vary greatly between different programming languages,
+there is no one-size-fits-all approach for ingesting your logs into Elasticsearch.
+The following tips should hopefully get you going in the right direction:
+
+**Download Filebeat**
+
+There are many ways to download and get started with Filebeat.
+Read the {filebeat-ref}/filebeat-installation.html[Filebeat Installation] guide to determine which is best for you.
+
+**Configure Filebeat**
+
+Modify the {filebeat-ref}/filebeat-configuration.html[`filebeat.yml`] configuration file to your needs.
+Here are some recommendations:
+
+* Set `filebeat.inputs` to point to the source of your logs
+* Point Filebeat to the same Elastic Stack that is receiving your APM data
+  * If you're using Elastic cloud, set `cloud.id` and `cloud.auth`.
+  * If your using a manual setup, use `output.elasticsearch.hosts`.
+
+[source,yml]
+----
+filebeat.inputs:
+- type: log <1>
+  paths: <2>
+    - /var/log/*.log
+cloud.id: "staging:dXMtZWFzdC0xLmF3cy5mb3VuZC5pbyRjZWMNjN2Q3YTllOTYyNTc0Mw==" <3>
+cloud.auth: "elastic:YOUR_PASSWORD" <4>
+----
+<1> Configures the `log` input
+<2> Path(s) that must be crawled to fetch the log lines
+<3> Used to resolve the Elasticsearch and Kibana URLs for Elastic Cloud
+<4> Authorization token for Elastic Cloud
+
+**JSON logs**
+
+For JSON logs you can use the {filebeat-ref}/filebeat-input-log.html[`log` input] to read lines from log files.
+Here's what a sample configuration might look like:
+
+[source,yml]
+----
+filebeat.inputs:
+  json.keys_under_root: true <1>
+  json.add_error_key: true <2>
+  json.message_key: message <3>
```

Comment:
I don't really understand (even after reading the Filebeat docs) what this is used for. I think we can safely omit this.

Reply:
`json.message_key: message`? It's also used here: https://github.com/elastic/apm-integration-testing/blame/ec0548b4cc674167be2b1adf78acc3d950df7f8b/docker/filebeat/filebeat.yml#L37. Perhaps @kuisathaverat can explain?

Reply:
It was really added here https://github.com/elastic/apm-integration-testing/commit/e64dc7c85ec83255724ed10c280f89dc5e5b1367#diff-b24360abd963736d9f4a0abfd0c68056 it simply tells that the key `message` would contain some text message, it is difficult to understand if you never saw the JSON Docker output log format
  

```
{"type": "server", "timestamp": "2019-09-23T11:06:15,953+0000", "level": "INFO", "component": "o.e.d.DiscoveryModule", "cluster.name": "docker-cluster", "node.name": "0f7a3616c26e",  "message": "using discovery type [single-node] and seed hosts providers [settings]"  }
```